subworkflows/local/locidex.nf

```diff
@@ -0,0 +1,63 @@
+include { LOCIDEX_EXTRACT } from "../../modules/local/locidex_extract.nf"
+include { LOCIDEX_SEARCH } from "../../modules/local/locidex_search.nf"
+include { LOCIDEX_REPORT } from "../../modules/local/locidex_report.nf"
+
+
+
+workflow LOCIDEX {
+    take:
+    contigs // val(meta), path(contigs)
+    top_hit // val(meta), top_hit
+
+    main:
+    reports = Channel.empty()
+    versions = Channel.empty()
+
+    paired_species = top_hit.join(contigs)
+
+    paired_dbs =  paired_species.map{
+        meta, top_hit, contigs -> tuple(meta, contigs, id_scheme(top_hit))
+    }.branch{
+        paired: it[2]
+        fallthrough: true
+    }
+
+    // TODO add to reports the database for allele calls to report
+    paired_dbs.fallthrough.subscribe {
+        log.info "No allele scheme identified for ${it[0].id}."
+    }
+
+    extracted_lx = LOCIDEX_EXTRACT(paired_dbs.paired)
+    versions = versions.mix(extracted_lx.versions)
+
+    allele_calls = LOCIDEX_SEARCH(extracted_lx.extracted_seqs)
+    versios = versions.mix(allele_calls.versions)
+
+    report_lx = LOCIDEX_REPORT(allele_calls.allele_calls)
+    versions = versions.mix(report_lx.versions)
+
+    emit:
+    versions
+
+
+}
+
+def id_scheme(top_hit){
+    /* Pick the correct allele scheme based off of the species top-hit
+    */
+
+    def selected_scheme = params.allele_scheme
+    if(selected_scheme){
+        return selected_scheme
+    }
+
+    for( scheme in params.locidex.schemes){
+        search_param = scheme.value.search.search
+        if(top_hit.contains(search_param)){
+            selected_scheme = scheme.value.db ? file(scheme.value.db) : false // Need a falue present that will send the scheme to the fallthrough case
```

Comment:
Does `null` not work for the fallthrough case? Could you maybe change that to work with a return of `null`?

The issue with returning `false` is that you are mixing return types of the function. That is when it finds a scheme it will return a file, but if it doesn't find a scheme it returns a boolean.

Reply:
yes null will work! I thought it would be better to be explicit!

Reply:
updated in:  8aaba2b